great_expectations/util.py

```diff
@@ -1437,19 +1440,51 @@ def is_ndarray_datetime_dtype(
 
 
 def convert_ndarray_to_datetime_dtype_best_effort(
-    data: np.ndarray, fuzzy: bool = False
-) -> np.ndarray:
+    data: np.ndarray, parse_strings_as_datetimes: bool = False, fuzzy: bool = False
+) -> Tuple[bool, bool, np.ndarray]:
     """
     Attempt to parse all elements of 1-D "np.ndarray" argument into "datetime.datetime" type objects.
+
+    Returns:
+        Boolean flag -- True if all elements of original "data" were "datetime.datetime" type objects; False, otherwise.
+        Boolean flag -- True, if conversion was performed; False, otherwise.
+        Output "np.ndarray" (converted, if necessary).
     """
+    if is_ndarray_datetime_dtype(
+        data=data, parse_strings_as_datetimes=False, fuzzy=fuzzy
+    ):
+        return True, False, data
+
     value: Any
-    if all(is_parseable_date(value=value) for value in data):
+    if is_ndarray_datetime_dtype(
+        data=data, parse_strings_as_datetimes=parse_strings_as_datetimes, fuzzy=fuzzy
+    ):
         try:
-            return np.asarray([parse(value, fuzzy=fuzzy) for value in data])
+            return (
+                False,
+                True,
+                np.asarray([parse(value, fuzzy=fuzzy) for value in data]),
+            )
         except (TypeError, ValueError):
-            return data
+            pass
+
+    return False, False, data
 
-    return data
+
+def convert_ndarray_datetime_to_float_dtype(data: np.ndarray) -> np.ndarray:
+    """
+    Conver all elements of 1-D "np.ndarray" argument from "datetime.datetime" type to "timestamp" "float" type objects.
+    """
+    value: Any
+    return np.asarray([value.timestamp() for value in data])
+
+
+def convert_ndarray_float_to_datetime_dtype(data: np.ndarray) -> np.ndarray:
+    """
+    Conver all elements of 1-D "np.ndarray" argument from "float" type to "datetime.datetime" type objects.
```

Comment:
"Convert"

Reply:
@NathanFarmer Thank you for the thoroughness!  Sorry!  Fixing in next commit.